checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,308 @@
+import org.checkerframework.checker.mustcall.CreatesObligationElementSupplier;
+import org.checkerframework.checker.mustcall.MustCallAnnotatedTypeFactory;
+import org.checkerframework.checker.mustcall.MustCallChecker;
+import org.checkerframework.checker.mustcall.MustCallNoCreatesObligationChecker;
+import org.checkerframework.checker.mustcall.qual.CreatesObligation;
+import org.checkerframework.checker.mustcall.qual.MustCall;
+import org.checkerframework.checker.mustcall.qual.MustCallAlias;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.checker.resourceleak.MustCallInvokedChecker.LocalVarWithTree;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.dataflow.cfg.ControlFlowGraph;
+import org.checkerframework.dataflow.cfg.node.LocalVariableNode;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.expression.LocalVariable;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreeUtils;
+import org.checkerframework.javacutil.TypesUtils;
+
+public class ResourceLeakAnnotatedTypeFactory extends CalledMethodsAnnotatedTypeFactory
+    implements CreatesObligationElementSupplier {
+
+  final ExecutableElement calledMethodsValueElement =
+      TreeUtils.getMethod(CalledMethods.class, "value", 0, processingEnv);
+
+  final ExecutableElement mustCallValueElement =
+      TreeUtils.getMethod(MustCall.class, "value", 0, processingEnv);
+
+  final ExecutableElement ensuresCalledMethodsVarArgsValueElement =
+      TreeUtils.getMethod(EnsuresCalledMethodsVarArgs.class, "value", 0, processingEnv);
+
+  final ExecutableElement ensuresCalledMethodsValueElement =
+      TreeUtils.getMethod(EnsuresCalledMethods.class, "value", 0, processingEnv);
+
+  final ExecutableElement ensuresCalledMethodsMethodsElement =
+      TreeUtils.getMethod(EnsuresCalledMethods.class, "methods", 0, processingEnv);
+
+  /** The CreatesObligation.List.value field/element. */
+  private final ExecutableElement createsObligationListValueElement =
+      TreeUtils.getMethod(CreatesObligation.List.class, "value", 0, processingEnv);
+
+  /** The CreatesObligation.value field/element. */
+  private final ExecutableElement createsObligationValueElement =
+      TreeUtils.getMethod(CreatesObligation.class, "value", 0, processingEnv);
+
+  /**
+   * Bidirectional map to preserve temporary variables created for nodes with non-empty @MustCall
+   * annotation and the corresponding nodes.
+   */
+  private BiMap<LocalVariableNode, Tree> tempVarToNode = HashBiMap.create();
```

Comment:
@kelloggm @Nargeshdb does this map and the corresponding accessor methods belong in the `MustCallAnnotatedTypeFactory`?  We moved logic for creating temp variables there, so I wonder if it'd be cleaner to put this there as well